src/code/graph.c

```diff
@@ -141,6 +141,9 @@ void Graph_InitTHGA(GraphicsContext* gfxCtx) {
     gfxCtx->overlayBuffer = pool->overlayBuffer;
     gfxCtx->workBuffer = pool->workBuffer;
 
+    //! @bug fbIdx is a signed integer that can overflow into the negatives. When compiled with IDO, the remainder
+    //! operator will yield -1 for odd negative values of fbIdx (i.e. the same as C99 onwards).
+    //! This results in an out of bounds array access in SysCfb_GetFbPtr due to the negative index value.
     gfxCtx->curFrameBuffer = SysCfb_GetFbPtr(gfxCtx->fbIdx % 2);
```

Comment:
feel like we are missing more here. This sounds like a problem even with IDO. What are the consequences, if any?

Reply:
afaict fbIdx is only reset (to 0) when the game boots, in `Graph_Init`
so when 2^31 frames have elapsed since boot overflow happens

at 60fps that'd take `2^31/60/3600/24 = 414 days`

as for what would happen, it seems like `sSysCfbFbPtr[-1]` is `sSpeedMeterTimeEntryPtr + 4` (on gc-eu-mq-dbg and ntsc-1.0 at least), which is bss padding in speed_meter.o. So it would return NULL
```
$ ./sym_info.py sSysCfbFbPtr
Symbol 'sSysCfbFbPtr' (VRAM: 0x8016A590, VROM: None, SIZE: 0x8, build/gc-eu-mq-dbg/src/code/sys_cfb.o)
$ ./sym_info.py 0x8016A58C
0x8016A58C is at 0x4 bytes inside 'sSpeedMeterTimeEntryPtr' (VRAM: 0x8016A588, VROM: None, SIZE: 0x8, build/gc-eu-mq-dbg/src/code/speed_meter.o)

$ ./sym_info.py -v ntsc-1.0 sSysCfbFbPtr
Symbol 'sSysCfbFbPtr' (VRAM: 0x80120FD0, VROM: None, SIZE: 0x8, build/ntsc-1.0/src/code/sys_cfb.o)
$ ./sym_info.py -v ntsc-1.0 0x80120FCC
0x80120FCC is at 0x4 bytes inside 'sSpeedMeterTimeEntryPtr' (VRAM: 0x80120FC8, VROM: None, SIZE: 0x8, build/ntsc-1.0/src/code/speed_meter.o)
```

as for what would using a NULL framebuffer result in, I'm not sure

Reply:
I see. The comment on how long it would take is very helpful. as for seeing that result, i guess i can try to change the code to simulate it

Reply:
setting the index to -1 made the game crash but with no crash screen. it triggered the following print/error state in graph.c `PRINTF(T("RCPが帰ってきませんでした。", "RCP did not return."));`

Reply:
I agree on removing "IDO", integer overflow is technically UB but every compiler should behave the same way here

Reply:
I think my concerns is that the current wording it implies that only IDO has this behavior. How about something like "compiler with IDO or a modern C compiler (e.g. gcc)"?

Reply:
if its related to the C standard then shouldnt it just say that, as opposed to mentioning any compiler?

Reply:
> if its related to the C standard then shouldnt it just say that, as opposed to mentioning any compiler?

I feel that would just lead to wording that makes things more confusing overall. Because we compile for C89, it is incorrect to assume that `-1 % 2 is -1`; it can be both -1 (which triggers the bug) and 1 (which exhibits the intended behavior). If you don't explain that IDO implements a remainder operation compliant with C99 (which triggers the bug), how would you explain that the bug exists in retail builds?

To me, it just seems simpler to state the behavior for C99+ and IDO than go into the weeds about the C89 standard here.

Reply:
```suggestion
    //! @bug fbIdx is a signed integer that can overflow into the negatives. When compiled with a C99+ compiler or IDO, the remainder
    //! operator will yield -1 for odd negative values of fbIdx.
    //! This results in an out of bounds array access in SysCfb_GetFbPtr due to the negative index value.
    gfxCtx->curFrameBuffer = SysCfb_GetFbPtr(gfxCtx->fbIdx % 2);
```
? (modulo formatting)